File: SEPTEMBER-2025/02-09-2025/Swap Kth nodes from ends.py
'''
class Node:
    def __init__(self, x):
        self.data = x
        self.next = None
'''
class Solution:
    def swapKth(self, head, k):
        # code here
        first=head
        s1=None
        s2=head
        for i in range(k):
            if i==k-1:
                s1=first
            if not first:
                return head
            first=first.next
        while first:
            s2=s2.next
            first=first.next
        s1.data, s2.data=s2.data, s1.data
        return head

File: SEPTEMBER-2025/02-09-2025/Swap Kth nodes from ends.cpp
class Solution {
  public:
    Node* swapKth(Node* head, int k) {
        // code here
        Node * temp = head;
        int count = 1;   
        while(temp->next!= NULL){
            count++;
            temp = temp->next;
        } 
        if(count<k) return head; 
        temp = head;
        for(int i=1; i<k; i++){
            temp = temp->next;
        }     
        Node * a = head;
        for(int i=1; i < count-k+1; i++){
            a = a->next;
        }        
        int tempdata = a->data;
        a->data = temp->data;
        temp->data = tempdata;
        return head;
    }
};